src/lsp/sql_parser/grammar.mly

```diff
@@ -0,0 +1,452 @@
+%{
+(**************************************************************************)
+(*                                                                        *)
+(*                        SuperBOL OSS Studio                             *)
+(*                                                                        *)
+(*  Copyright (c) 2022-2023 OCamlPro SAS                                  *)
+(*                                                                        *)
+(* All rights reserved.                                                   *)
+(* This source code is licensed under the GNU Affero General Public       *)
+(* License version 3 found in the LICENSE.md file in the root directory   *)
+(* of this source tree.                                                   *)
+(*                                                                        *)
+(**************************************************************************)
+open Sql_ast
+open Cobol_common.Srcloc.INFIX
+module Overlay_manager = Sql_overlay_manager
+
+%}
+%token COLON COMMA SEMICOLON EQUAL PLUS MINUS OR AND IN
+%token LESS_EQ LESS GREAT_EQ GREAT DIFF
+%token <char> SPECHAR
+%token EOF
+%token EXEC SQL END_EXEC
+%token BEGIN END DECLARE SECTION DISCONNECT LPAR RPAR DOT STAR COUNT_STAR
+(*For whenever*)
+%token WHENEVER NOT FOUND SQLERROR SQLWARNING CONTINUE PERFORM GOTO GO
+(*For include*)
+%token INCLUDE
+(*For Connect*)
+%token CONNECT TO USER IDENTIFIED BY AT USING RESET AS
+(*For Transaction*)
+%token START TRANSACTION
+(*For Rollback*)
+%token ROLLBACK WORK RELEASE SAVEPOINT 
+(*For commit*)
+%token COMMIT
+(*FOr other esql with opt at*)
+%token SELECT INTO STATEMENT CURSOR FOR PREPARE FROM EXECUTE IMMEDIATE WITH HOLD
+%token UNION EXCEPT INTERSECT
+%token INSERT OPEN CLOSE FETCH DELETE UPDATE SET IGNORE ALL OF WHERE CURRENT TABLE
+%token WHEN ORDER VALUES IS NULL DEFAULT GROUP HAVING
+%token JOIN INNER NATURAL LEFT RIGHT OUTER ON BETWEEN
+(*Sort by*)
+%token DESC ASC
+(*types*)
+%token VARCHAR DATE INTEGER TIMESTAMP
+(*exeptions*)
+%token THEN RAISE EXCEPTION
+%token <string> TOKEN
```

Comment:
Not sure `TOKEN` is a proper name for a specific token. In COBOL we have `WORD`, maybe there's an equivalent in SQL?